bee-common/src/worker.rs

```diff
@@ -0,0 +1,23 @@
+// Copyright 2020 IOTA Stiftung
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+// the License. You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+// an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and limitations under the License.
+
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum Error {
+    /// A wrapper for an async task error.
+    #[error("An asynchronous operation failed.")]
+    AsynchronousOperationFailed(#[from] async_std::io::Error),
```

Comment:
This shouldn't be tied to `async_std` anymore. This is user's responsibility to map to `AsynchronousOperationFailed`.

Reply:
yea, you're right :slightly_smiling_face: